netx/dialer/http3dns_test.go

```diff
@@ -127,11 +128,11 @@ func TestHTTP3DNSDialerDialEarlyFails(t *testing.T) {
 		NextProtos: []string{"h3-29"},
 	}
 	expected := errors.New("mocked DialEarly error")
-	mockDialEarly := func(net.PacketConn, net.Addr, string, *tls.Config, *quic.Config) (quic.EarlySession, error) {
+	mockDialEarly := func(context.Context, net.PacketConn, net.Addr, string, *tls.Config, *quic.Config) (quic.EarlySession, error) {
```

Comment:
```suggestion
	mockDialEarly := func(context.Background(), net.PacketConn, net.Addr, string, *tls.Config, *quic.Config) (quic.EarlySession, error) {
```

I think we should pass in a background context here?

Reply:
I'm not exactly sure what you mean, mockDialEarly is called with context.Background() a few lines later, this is just the declaration. Or what am I missing here? :)

Reply:
You're not missing anything, I think I've read the diff incorrectly :-)